### PR TITLE
Fix hash functions for IPv4

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Follow-up fix for Replace domain IP types (IPv4, IPv6) with native https://github.com/ClickHouse/ClickHouse/pull/43221

### CODE REVIEW

File: tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+javaHashUTF16LE(toIPv4('1.2.3.4')):          24190
+intHash64(toIPv4('1.2.3.4')):                5715546585361069049
+intHash32(toIPv4('1.2.3.4')):                3152671896
+metroHash64(toIPv4('1.2.3.4')):              5715546585361069049
+farmHash64(toIPv4('1.2.3.4')):               5715546585361069049
+hex(sipHash128(toIPv4('1.2.3.4'))):          DBB6A76B92B59789EFB42185DC32311D
+hex(murmurHash3_128(toIPv4('1.2.3.4'))):     549E9EF692591F6BB55874EF9A0DE88E
+jumpConsistentHash(toIPv4('1.2.3.4'), 42):   37
+sipHash64(toIPv4('1.2.3.4')):                10711397536826262068
+kostikConsistentHash(toIPv4('1.2.3.4'), 42): 0
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference:17]: Cosmetic: Looking at #35553, `yandexConsistentHash` was renamed to `kostikConsistentHash`, you could include them side-by-side in the test (or simply omit `yandexConsistentHashing`).

EDIT: Somehow, `kostikConsistentHash` isn't documented [here](https://clickhouse.com/docs/en/sql-reference/functions/hash-functions/). Perhaps you could at least include a mention of it, see "Functions/kostikConsistentHash.cpp".

File: tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference [hunk 2]
```diff
@@ -0,0 +1,43 @@
+Row 1:
+──────
+ipv4:                                        1.2.3.4
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference:3]: BLAKE, MD*() and SHA*() are missing. Intended?

File: src/Functions/FunctionsConsistentHashing.h [hunk 3]
```diff
@@ -43,7 +44,7 @@ class FunctionConsistentHashImpl : public IFunction
 
     DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
     {
-        if (!isInteger(arguments[0]))
+        if (!isInteger(arguments[0]) && !isIPv4(arguments[0]))
```
Comment (rschu1ze) [inline src/Functions/FunctionsConsistentHashing.h:47]: + ` && !isIPv6(arguments[0])`?

Comment (yakov-olkhovskiy) [reply src/Functions/FunctionsConsistentHashing.h:47]: I made IPv6 and IPv4 to be represented by Numper, Integer and UnsignedInteger - see DataTypeIPv4andIPv6.h - for IPv4 it's definitely true, but not sure about IPv6...

**Comment by yakov-olkhovskiy** [comment]
farmHash64 seems to be not stable for strings

**Comment by yakov-olkhovskiy** [comment]
indeed:
https://github.com/ClickHouse/ClickHouse/blob/master/contrib/libfarmhash/farmhash.cc#L1894-L1903

File: tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference (hunk 2 above)
Comment (den-crane) [reply tests/queries/0_stateless/02184_hash_functions_and_ip_types.reference:3]: Not intended. 
I've used `select * from system.functions where name ilike '%hash%';` to find hash functions and it obviously was not enough.

**Review by rschu1ze — APPROVED** [review]

**Comment by rschu1ze** [comment]
just merge whenever you think it's ready to merge

**Comment by yakov-olkhovskiy** [comment]
BLACK3 hash seems also unstable - in msan build it treats parameter as an array, but release build treats parameter as a null-terminated string... I'm not sure how critical it is, since according to docs it takes string parameter, but it makes difference for FixedString type.
cc @alexey-milovidov @KochetovNicolai

**Pull request merged**